lib/internal/async_hooks.js

```diff
@@ -243,13 +245,84 @@ function restoreActiveHooks() {
   active_hooks.tmp_fields = null;
 }
 
+const triggerAsyncIdSymbol = Symbol('triggerAsyncId');
+const destroyedSymbol = Symbol('destroyed');
+
+function wrapPromise(promise, parent, silent) {
+  const asyncId = getOrSetAsyncId(promise);
+
+  if (parent) {
+    promise[triggerAsyncIdSymbol] = getOrSetAsyncId(parent);
+    promise.isChainedPromise = true;
+  } else {
+    promise[triggerAsyncIdSymbol] = getDefaultTriggerAsyncId();
+    promise.isChainedPromise = false;
+  }
+
+  if (!silent && initHooksExist()) {
+    emitInitScript(asyncId, 'PROMISE', promise[triggerAsyncIdSymbol], promise);
+  }
+
+  if (destroyHooksExist()) {
+    const destroyed = { destroyed: false };
+    promise[destroyedSymbol] = destroyed;
+    registerDestroyHook(promise, asyncId, destroyed);
+  }
+}
+
+function promiseHook(type, promise, parent) {
+  if (!promise) return;
```

Comment:
How could this happen?

Reply:
Just some leftover stuff from when I was debugging something. Should get removed soon.